src/Semantic/Task.hs

```diff
@@ -269,6 +269,12 @@ runParser blob@Blob{..} parser = case parser of
       parseToAST (configTreeSitterParseTimeout config) language blob
         >>= maybeM (throwError (SomeException ParserTimedOut))
 
+  UnmarshalParser language ->
+    time "parse.tree_sitter_ast_parse" languageTag $ do
+      config <- asks config
+      parseToPreciseAST (configTreeSitterParseTimeout config) language blob
+        >>= maybeM (throwError (SomeException ParserTimedOut))
```

Comment:
We are incorrectly reporting a couple of different classes of problems as timeouts, as indeed we are for `ASTParser`, above. I don’t know that we’re actually running into non-timeout `Nothing` values that we’re reporting as timeouts, but I also don’t know that we’re not.